yt_dlp/extractor/mediaset.py

```diff
@@ -179,13 +237,24 @@ def _real_extract(self, url):
                         'assetTypes': asset_type,
                     }), guid, 'Downloading %s SMIL data' % (f.split('+')[0]))
             except ExtractorError as e:
+                if not geo_e and isinstance(e, GeoRestrictedError):
+                    geo_e = e
                 if not first_e:
                     first_e = e
-                break
-            formats.extend(tp_formats)
+                continue
+            tp_formats = self._check_drm_formats(tp_formats, guid)
+            if tp_formats == 'DRM':
+                has_drm = True
+            else:
+                formats.extend(tp_formats)
             subtitles = self._merge_subtitles(subtitles, tp_subtitles)
-        if first_e and not formats:
-            raise first_e
+
+        # check for errors and report them
+        if has_drm and not formats:
+            self.report_drm(guid)
```

Comment:
Most of the time, you dont need to raise DRM error manually. leave the formats with DRM in the format list with a key `has_drm=True`. Then the core program will automatically raise the appropriate error.

With appropriate changes to `_check_drm_formats` (see my other comment), this should simplify to
```py
            self._check_drm_formats(tp_formats, guid)
            formats.extend(tp_formats)
            subtitles = self._merge_subtitles(subtitles, tp_subtitles)
```

Reply:
ok